scripts/generate_docs

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+#
+# Check that Rust documentation is generated without any warnings.
+
+readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
+# shellcheck source=scripts/common
+source "$SCRIPTS_DIR/common"
+
+if cargo doc --document-private-items --no-deps 2>&1 | grep -iq -e '^WARNING'; then
```

Comment:
Perhaps add a comment explaining why we need this script, i.e. that warnings cannot be promoted to errors via cargo, or something like that.